core/src/main/scala/cats/instances/set.scala

```diff
@@ -1,56 +1,17 @@
 package cats
 package instances
 
-import scala.annotation.tailrec
-
 import cats.syntax.show._
 
 trait SetInstances extends cats.kernel.instances.SetInstances {
 
-  implicit val catsStdInstancesForSet: Foldable[Set] with MonoidK[Set] =
-    new Foldable[Set] with MonoidK[Set] {
+  implicit val catsStdInstancesForSet: MonoidK[Set] =
+    new MonoidK[Set] {
 
       def empty[A]: Set[A] = Set.empty[A]
 
       def combineK[A](x: Set[A], y: Set[A]): Set[A] = x | y
 
-      def foldLeft[A, B](fa: Set[A], b: B)(f: (B, A) => B): B =
```

Comment:
Same concern here about adding a law that this violates. Maybe a == b implies toList(a) == toList(b)

Reply:
We're going to need to spend some more time to find something. That one won't work, because that way e.g
`Either.Left(123) =!= Either.Left(-30)` would imply `List() =!= List()`

Reply:
I'm not sure if it's the best solution but I worked around it, by only considering the positive case. I.e `a == b` implies `toList(a) == toList(b)` but not `a != b` implies `toList(a) != toList(b)`

Reply:
This is what I meant (which is why I didn’t say (a == b) == (a.toList == b.toList))